CI/update-autocompletion.lua

```diff
@@ -0,0 +1,70 @@
+local http_request = require "http.request"
+local lunajson = require "lunajson"
+
+local function trim(s)
+  return (s:gsub("^%s*(.-)%s*$", "%1"))
+end
+
+local function magiclines(s)
+  if s:sub(-1)~="\n" then s=s.."\n" end
+  return s:gmatch("(.-)\n")
+end
+
+local function scrapeLuaFunctions(htmlbody)
+  local funcs = {}
+  local funcsHash = {}
+  local count = 0
+  local state = 0
+  local line = nil
+  local match = nil
+  local name, usage, definition
+  for line in magiclines(htmlbody) do
+    if state == 0 then
+      --print("testing match on " .. line)
+      name = string.match(line, '<h2><span class="mw%-headline" id="(.-)">.-</span></h2>')
+      if name then
+        state = 1
+        --print("Name: " .. name)
+      end
+    elseif state == 1 then
+      usage = string.match(line, '<dl><dt>(.-)</dt>')
+      if usage then
+        state = 0
+        --print("Usage: " .. usage)
+        func = {}
+        func.name = trim(name)
+        func.usage = trim(usage)
+        table.insert(funcs, func)
+      end
+    end
+  end
+
+  table.sort(funcs, function (a,b) return a.name < b.name end)
```

Comment:
Why is the sorting here? I mean in the end, we simply put them into a named table/hash anyways.

Reply:
Doesn't make sense to me either. The original script did it this way - I'll take it out.